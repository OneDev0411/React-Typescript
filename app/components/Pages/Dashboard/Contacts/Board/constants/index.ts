export const Tags = ['Warm', 'Hot', 'Past Client']

export const Columns = [
  {
    title: 'All Contacts',
    tag: undefined
  },
  {
    title: Tags[0],
    tag: Tags[0]
  },
  {
    title: Tags[1],
    tag: Tags[1]
  },
  {
    title: Tags[2],
    tag: Tags[2]
  }
]
