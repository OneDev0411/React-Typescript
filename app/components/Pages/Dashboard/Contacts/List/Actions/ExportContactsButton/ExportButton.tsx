import React from 'react'

import { List, useTheme } from '@material-ui/core'
import {
  mdiAt,
  mdiProgressDownload,
  mdiEmailOutline,
  mdiExportVariant
} from '@mdi/js'

import { GridActionButton } from '@app/views/components/Grid/Table/features/Actions/Button'
import { BaseDropdown } from 'components/BaseDropdown'
import { SvgIcon } from 'components/SvgIcons/SvgIcon'

import { Item } from './Item'
import { DownloadType, ExportType } from './types'

interface Props {
  disabled: boolean
  onExportClick: (exportType: ExportType) => Promise<void>
}

export function ExportButton({ disabled, onExportClick }: Props) {
  const theme = useTheme()
  const iconColor = theme.palette.common.black

  const items: DownloadType[] = [
    {
      title: 'Export For Mailers',
      description:
        'Use this for print campaigns, like importing into XpressDocs',
      type: ExportType.Mailer,
      icon: <SvgIcon path={mdiEmailOutline} color={iconColor} />
    },
    {
      title: 'Export For Email',
      description: 'Use this for email campaigns, like importing into Rezora',
      type: ExportType.Email,
      icon: <SvgIcon path={mdiAt} color={iconColor} />
    },
    {
      title: 'Full Export',
      description: 'Download everything, all your contact fields',
      type: ExportType.Full,
      icon: <SvgIcon path={mdiProgressDownload} color={iconColor} />
    }
  ]

  const onClickItem = async (exportType: ExportType, onClose: () => void) => {
    onExportClick(exportType)
    onClose()
  }

  return (
    <BaseDropdown
      renderDropdownButton={({ ref, onClick }) => (
        <GridActionButton
          attachedElementRef={ref}
          label="Export"
          icon={mdiExportVariant}
          disabled={disabled}
          onClick={onClick}
        />
      )}
      DropdownToggleButtonProps={{
        disabled,
        size: 'small',
        variant: 'outlined'
      }}
      renderMenu={({ close }) => (
        <List>
          {items.map(item => (
            <Item
              key={item.type}
              icon={item.icon}
              title={item.title}
              onClick={() => onClickItem(item.type, close)}
              description={item.description}
            />
          ))}
        </List>
      )}
    />
  )
}
