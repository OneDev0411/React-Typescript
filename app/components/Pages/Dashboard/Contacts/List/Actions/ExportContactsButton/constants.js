export const SAME_ROW_DOWNLOAD_TYPE = 'same'
export const SEPARATE_ROW_DOWNLOAD_TYPE = 'separate'
