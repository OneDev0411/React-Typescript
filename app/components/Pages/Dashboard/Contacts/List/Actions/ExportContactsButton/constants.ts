import IconMailer from 'components/SvgIcons/Export/IconMailer'
import IconEmail from 'components/SvgIcons/Export/IconEmail'
import IconFull from 'components/SvgIcons/Export/IconFull'

export const FULL_EXPORT_TYPE = 'contact_export'
export const EMAIL_EXPORT_TYPE = 'contact_email_export'
export const MAILER_EXPORT_TYPE = 'contact_joint_export'

enum ExportType {
  Full = 'contact_export',
  Email = 'contact_email_export',
  Mailer = 'contact_joint_export'
}

export interface DownloadType {
  title: string
  description: string
  type: ExportType
  icon: React.ReactNode
}

export const DOWNLOAD_TYPES_DROPDOWN_ITEMS: DownloadType[] = [
  {
    title: 'Export For Mailers',
    description: 'Use this for print campaigns, like importing into XpressDocs',
    type: ExportType.Mailer,
    icon: IconMailer
  },
  {
    title: 'Export For Email',
    description: 'Use this for email campaigns, like importing into Rezora',
    type: ExportType.Email,
    icon: IconEmail
  },
  {
    title: 'Full Export',
    description: 'Download everything, all your contact fields',
    type: ExportType.Full,
    icon: IconFull
  }
]
