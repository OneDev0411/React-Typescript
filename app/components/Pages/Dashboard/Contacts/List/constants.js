export const SORT_FIELD_SETTING_KEY = 'grid_contacts_sort_field'

export const OPEN_HOUSE_FILTER_ID = 'openHouse'
export const FLOW_FILTER_ID = 'flow'

export const ORIGINS = [
  {
    label: 'Brokerage widget',
    value: 'BrokerageWidget'
  },
  {
    label: 'Created by you',
    value: 'ExplicitlyCreated'
  },
  {
    label: 'iOS Contact',
    value: 'IOSAddressBook'
  },
  {
    label: 'Rechat Contact',
    value: 'SharesRoom'
  },
  {
    label: 'Outlook',
    value: 'External/Outlook'
  },
  {
    label: 'Open House',
    value: 'OpenHouse'
  },
  {
    label: 'CSV',
    value: 'CSV'
  }
]
