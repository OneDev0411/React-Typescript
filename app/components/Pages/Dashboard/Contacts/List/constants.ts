export const SORT_FIELD_SETTING_KEY = 'grid_contacts_sort_field'
export const VIEW_MODE_FIELD_SETTING_KEY = 'contact_view_mode_field'

export const OPEN_HOUSE_FILTER_ID = 'openHouse'
export const FLOW_FILTER_ID = 'flow'

export const IMPORT_TOOLTIP_VISITED_SETTINGS_KEY = 'import_tooltip_visited'

export const PARKED_CONTACTS_LAST_SEEN_SETTINGS_KEY =
  'parked-contacts-last-seen'

export const PARKED_CONTACTS_LIST_ID = 'parked'

export const DUPLICATE_CONTACTS_LIST_ID = 'duplicates'
export const CONTACT_CHUNK_COUNT = 50

export const ORIGINS = [
  {
    label: 'Brokerage widget',
    description: 'Created from a brokerage widget',
    value: 'BrokerageWidget'
  },
  {
    label: 'Created by you',
    description: 'Created by you',
    value: 'ExplicitlyCreated'
  },
  {
    label: 'iOS Contact',
    description: 'From your iOS address book',
    value: 'IOSAddressBook'
  },
  {
    label: 'Rechat Contact',
    description: 'Created because you share a room',
    value: 'SharesRoom'
  },
  {
    label: 'Outlook',
    description: 'Imported from Outlook',
    value: 'Microsoft'
  },
  {
    label: 'Open House',
    description: 'Created from an open house registration',
    value: 'OpenHouse'
  },
  {
    label: 'CSV',
    description: 'Imported from CSV',
    value: 'CSV'
  },
  {
    label: 'Google',
    description: 'Imported from Google',
    value: 'Google'
  },
  {
    label: 'Agent Network',
    description: 'Imported from Agent Network',
    value: 'AgentNetwork'
  },
  {
    label: 'Showing',
    description: 'Showing visitors',
    value: 'Showing'
  }
]
