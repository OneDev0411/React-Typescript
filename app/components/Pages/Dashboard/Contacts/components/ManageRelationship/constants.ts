export const frequencyOptions = {
  7: 'Every week',
  30: 'Every month',
  60: 'Every other month',
  90: 'Every three months',
  180: 'Every six months',
  365: 'Every year'
}
