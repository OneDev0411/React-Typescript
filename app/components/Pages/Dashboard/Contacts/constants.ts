export const CONTACTS_SEGMENT_NAME: string = 'contacts'

export const TOUCH_REMINDER_HINT_DISMISSED_SETTINGS_KEY =
  'contact_touch_reminder_hint_dismissed'
