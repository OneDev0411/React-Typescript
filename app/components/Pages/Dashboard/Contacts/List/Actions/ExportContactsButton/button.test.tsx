import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { DOWNLOAD_TYPES_DROPDOWN_ITEMS } from './constants'
import ExportButton from './button'

describe('ExportContactsButton inner button component', () => {
  it('renders', () => {
    const wrapper = shallow(
      <ExportButton disabled={false} onExportClick={() => null} />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders with proper items prop', () => {
    const wrapper = shallow(
      <ExportButton disabled={false} onExportClick={() => null} />
    )

    expect(wrapper.prop('items')).toBe(DOWNLOAD_TYPES_DROPDOWN_ITEMS)
  })
})
