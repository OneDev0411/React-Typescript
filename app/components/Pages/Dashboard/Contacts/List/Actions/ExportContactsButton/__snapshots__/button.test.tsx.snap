// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportContactsButton inner button component renders 1`] = `
<BasicDropdown
  buttonIcon={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "button__Xlsx-m7c26e-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          ";",
          "margin-right:0.5rem;",
        ],
      },
      "defaultProps": Object {
        "children": Array [
          <path
            d="M15.556 6.222h-1.334v-3.11a.445.445 0 0 0-.13-.315L11.425.13a.445.445 0 0 0-.314-.13H2.222a.444.444 0 0 0-.444.444v5.778H.444A.444.444 0 0 0 0 6.667v6.222c0 .245.199.444.444.444h1.334v2.223c0 .245.199.444.444.444h11.556a.444.444 0 0 0 .444-.444v-2.223h1.334A.444.444 0 0 0 16 12.89V6.667a.444.444 0 0 0-.444-.445zm-4-4.705l1.149 1.15h-1.15v-1.15zM2.666.89h8v2.667h2.667v2.666H2.667V.89zm10.668 14.22H2.667v-1.778h10.666v1.778zm1.778-2.667H.89V7.111h14.22v5.333z"
          />,
          <path
            d="M3.111 11.111c0-.338.168-.644.445-.827.276.183.444.49.444.827v.445h.889v-.445a1.87 1.87 0 0 0-.557-1.333c.35-.347.557-.82.557-1.334V8H4v.444a.987.987 0 0 1-.444.827.987.987 0 0 1-.445-.827V8h-.889v.444c0 .513.207.987.557 1.334a1.87 1.87 0 0 0-.557 1.333v.445h.89v-.445zm4.445-.444H6.222V8h-.889v3.556h2.223zm3.111-.445a.89.89 0 0 0-.89-.889H8.89c0-.245.2-.444.444-.444h1.334V8H9.333C8.598 8 8 8.598 8 9.333c0 .49.399.89.889.89h.889c0 .244-.2.444-.445.444H8v.889h1.333c.736 0 1.334-.598 1.334-1.334zm1.333.889c0-.338.168-.644.444-.827.277.183.445.49.445.827v.445h.889v-.445a1.87 1.87 0 0 0-.557-1.333c.35-.347.557-.82.557-1.334V8h-.89v.444a.987.987 0 0 1-.444.827.987.987 0 0 1-.444-.827V8h-.889v.444c0 .513.207.987.557 1.334a1.87 1.87 0 0 0-.557 1.333v.445H12v-.445z"
          />,
        ],
        "viewBox": "0 0 16 16",
      },
      "displayName": "button__Xlsx",
      "foldedComponentIds": Array [
        "XlsxIcon__Image-zvj0wa-0",
      ],
      "getDimensions": [Function],
      "getDimensionsCss": [Function],
      "render": [Function],
      "styledComponentId": "button__Xlsx-m7c26e-0",
      "target": "svg",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  buttonSize="small"
  buttonStyle={
    Object {
      "width": "inherit",
    }
  }
  buttonText="Export Contacts"
  disabled={false}
  itemRenderer={[Function]}
  items={
    Array [
      Object {
        "description": "Use this for print campaigns, like importing into XpressDocs",
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "IconMailer__Image-wyvm0b-0",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "defaultProps": Object {
            "children": <React.Fragment>
              <path
                d="M2.449 4h19.102C22.904 4 24 5.053 24 6.353v11.294c0 1.3-1.096 2.353-2.449 2.353H2.449C1.096 20 0 18.947 0 17.647V6.353C0 5.053 1.096 4 2.449 4zm0 1.882a.48.48 0 0 0-.49.47v11.295c0 .26.22.47.49.47h19.102c.27 0 .49-.21.49-.47V6.353a.48.48 0 0 0-.49-.47H2.449zm2.204 8.47c-.541 0-.98-.42-.98-.94s.439-.941.98-.941h3.674c.54 0 .98.421.98.94 0 .52-.44.942-.98.942H4.653zm0 2.824c-.541 0-.98-.42-.98-.94s.439-.942.98-.942h8.082c.54 0 .98.421.98.941s-.44.941-.98.941H4.653zM16.408 6.824h2.939c.947 0 1.714.737 1.714 1.647v2.823c0 .91-.767 1.647-1.714 1.647h-2.939c-.947 0-1.714-.737-1.714-1.647V8.471c0-.91.767-1.647 1.714-1.647zm.245 1.882v2.353h2.449V8.706h-2.449z"
              />
            </React.Fragment>,
            "fillColor": null,
            "fillColorRule": "&&& path, &&& use, &&& g",
            "size": null,
            "sizes": Object {},
            "viewBox": "0 0 24 24",
          },
          "displayName": "IconMailer",
          "foldedComponentIds": Array [],
          "getCss": [Function],
          "getDimensions": [Function],
          "propTypes": Object {
            "children": [Function],
            "fillColor": [Function],
            "fillColorRule": [Function],
            "size": [Function],
            "sizes": [Function],
            "viewBox": [Function],
          },
          "render": [Function],
          "styledComponentId": "IconMailer__Image-wyvm0b-0",
          "target": "svg",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "title": "Export For Mailers",
        "type": "contact_joint_export",
      },
      Object {
        "description": "Use this for email campaigns, like importing into Rezora",
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "IconEmail__Image-sc-199wmrd-0",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "defaultProps": Object {
            "children": <React.Fragment>
              <path
                d="M19.592 12v1.102a2.816 2.816 0 0 1-4.973 1.812A3.918 3.918 0 1 1 15.918 12v1.102a.857.857 0 0 0 1.715 0V12a5.634 5.634 0 1 0-3.13 5.047.98.98 0 0 1 .871 1.755A7.593 7.593 0 1 1 19.592 12zM12 24C5.373 24 0 18.627 0 12S5.373 0 12 0s12 5.373 12 12-5.373 12-12 12zm0-1.96c5.545 0 10.04-4.495 10.04-10.04 0-5.545-4.495-10.04-10.04-10.04C6.455 1.96 1.96 6.454 1.96 12c0 5.545 4.495 10.04 10.04 10.04zm0-8.08a1.96 1.96 0 1 0 0-3.92 1.96 1.96 0 0 0 0 3.92z"
              />
            </React.Fragment>,
            "fillColor": null,
            "fillColorRule": "&&& path, &&& use, &&& g",
            "size": null,
            "sizes": Object {},
            "viewBox": "0 0 24 24",
          },
          "displayName": "IconEmail",
          "foldedComponentIds": Array [],
          "getCss": [Function],
          "getDimensions": [Function],
          "propTypes": Object {
            "children": [Function],
            "fillColor": [Function],
            "fillColorRule": [Function],
            "size": [Function],
            "sizes": [Function],
            "viewBox": [Function],
          },
          "render": [Function],
          "styledComponentId": "IconEmail__Image-sc-199wmrd-0",
          "target": "svg",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "title": "Export For Email",
        "type": "contact_email_export",
      },
      Object {
        "description": "Download everything, all your contact fields",
        "icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "IconFull__Image-sc-1u3rnn8-0",
            "isStatic": false,
            "rules": Array [
              [Function],
            ],
          },
          "defaultProps": Object {
            "children": <React.Fragment>
              <path
                d="M10.977 10.37V.98c0-.541.458-.98 1.023-.98s1.023.439 1.023.98v9.39l1.323-1.267a1.056 1.056 0 0 1 1.447 0 .95.95 0 0 1 0 1.386l-3.07 2.938c-.4.383-1.047.383-1.447 0l-3.07-2.938a.95.95 0 0 1 0-1.386 1.056 1.056 0 0 1 1.448 0l1.323 1.267zm-7.93 5.059h17.906V8.327a.5.5 0 0 0-.511-.49h-3.07c-.565 0-1.023-.439-1.023-.98 0-.54.458-.98 1.023-.98h3.07c1.413 0 2.558 1.097 2.558 2.45v9.55c0 1.353-1.145 2.45-2.558 2.45h-7.419v1.714h2.814c.565 0 1.023.438 1.023.98 0 .54-.458.979-1.023.979H8.163c-.565 0-1.023-.439-1.023-.98 0-.54.458-.98 1.023-.98h2.814v-1.713H3.558C2.145 20.327 1 19.23 1 17.877v-9.55c0-1.353 1.145-2.45 2.558-2.45h3.07c.565 0 1.023.44 1.023.98 0 .541-.458.98-1.023.98h-3.07a.501.501 0 0 0-.511.49v7.102zm0 1.959v.49c0 .27.229.49.511.49h16.884c.282 0 .511-.22.511-.49v-.49H3.047z"
              />
            </React.Fragment>,
            "fillColor": null,
            "fillColorRule": "&&& path, &&& use, &&& g",
            "size": null,
            "sizes": Object {},
            "viewBox": "0 0 24 24",
          },
          "displayName": "IconFull",
          "foldedComponentIds": Array [],
          "getCss": [Function],
          "getDimensions": [Function],
          "propTypes": Object {
            "children": [Function],
            "fillColor": [Function],
            "fillColorRule": [Function],
            "size": [Function],
            "sizes": [Function],
            "viewBox": [Function],
          },
          "render": [Function],
          "styledComponentId": "IconFull__Image-sc-1u3rnn8-0",
          "target": "svg",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "title": "Full Export",
        "type": "contact_export",
      },
    ]
  }
  onChange={[Function]}
  selectedItem={null}
  style={
    Object {
      "display": "inline",
    }
  }
  upsideDown={true}
/>
`;
